autopush/tests/test_rs_integration.py

```diff
@@ -740,130 +709,278 @@ def test_delete_saved_notification(self):
-        exists, chans = yield deferToThread(
-            self.conn.db.message.all_channels,
-            client.uaid
-        )
-        assert exists is True
-        assert len(chans) == 1
-        yield deferToThread(
-            lm_message.save_channels,
-            client.uaid,
-            chans,
-        )
+        # Send an update with a properly formatted key.
+        yield client.send_notification(vapid=vapid)
 
-        # Remove the channels entry entirely from this month
-        yield deferToThread(
-            self.conn.db.message.table.delete_item,
-            Key={'uaid': client.uaid, 'chidmessageid': ' '}
-        )
+        # now try an invalid key.
+        new_key = ecdsa.SigningKey.generate(curve=ecdsa.NIST256p)
+        vapid = _get_vapid(new_key, claims)
 
-        # Verify the channel is gone
-        exists, chans = yield deferToThread(
-            self.conn.db.message.all_channels,
-            client.uaid,
-        )
-        assert exists is False
-        assert len(chans) == 0
+        yield client.send_notification(
+            vapid=vapid,
+            status=401)
 
-        # Send in a notification, verify it landed in last months notification
-        # table
-        data = uuid.uuid4().hex
-        with self.legacy_endpoint():
-            yield client.send_notification(data=data)
-        ts, notifs = yield deferToThread(lm_message.fetch_timestamp_messages,
-                                         uuid.UUID(client.uaid),
-                                         " ")
-        assert len(notifs) == 1
+        yield self.shut_down(client)
 
-        # Connect the client, verify the migration
+    @inlineCallbacks
+    def test_with_bad_key(self):
+        chid = str(uuid.uuid4())
+        client = Client("ws://localhost:{}/".format(connection_port))
         yield client.connect()
         yield client.hello()
+        result = yield client.register(chid=chid, key="af1883%&!@#*(",
+                                       status=400)
+        assert result["status"] == 400
 
-        # Pull down the notification
-        result = yield client.get_notification()
-        chan = client.channels.keys()[0]
-        assert result is not None
-        assert chan == result["channelID"]
+        yield self.shut_down(client)
 
-        # Acknowledge the notification, which triggers the migration
-        yield client.ack(chan, result["version"])
 
-        # Wait up to 4 seconds for the table rotation to occur
-        start = time.time()
-        while time.time()-start < 4:
-            c = yield deferToThread(
-                self.conn.db.router.get_uaid,
-                client.uaid)
-            if c["current_month"] == self.conn.db.current_msg_month:
-                break
-            else:
-                yield deferToThread(time.sleep, 0.2)
+class TestRustWebPushBroadcast(unittest.TestCase):
+    connection_port = 9052
+    endpoint_port = 9060
+    router_port = 9072
```

Comment:
nit: Base these as offsets from the globals? (looks like endpoint_port stays the same anyway)